packages/database/src/exp/Database.ts

```diff
@@ -0,0 +1,155 @@
+        this._internalDbUrl
+      );
+    }
+    return this._internalDb;
+  }
+
+  /**
+   * Modify this instance to communicate with the Realtime Database emulator.
+   *
+   * <p>Note: This method must be called before performing any other operation.
+   *
+   * @param host - the emulator host (ex: localhost)
+   * @param port - the emulator port (ex: 8080)
+   */
+  useEmulator(host: string, port: number): void {
+    this._delegate.useEmulator(host, port);
+  }
+
+  /**
+   * Returns a reference to the root or to the path specified in the provided
+   * argument.
+   *
+   * @param path - The relative string path or an existing Reference to a
+   * database location.
+   * @throws If a Reference is provided, throws if it does not belong to the
+   * same project.
+   * @returns Firebase reference.
+   */
+  ref(path?: string): Reference;
+  ref(path?: Reference): Reference;
+  ref(path?: string | Reference): Reference {
+    return typeof path === 'string'
+      ? this._delegate.ref(path)
+      : this._delegate.ref(path);
+  }
+
+  /**
+   * Returns a reference to the root or the path specified in url.
+   * We throw a exception if the url is not in the same domain as the
+   * current repo.
+   * @param url - A URL that refers to a database location.
+   * @returns A Firebase reference.
+   */
+  refFromURL(url: string): Reference {
+    return this._delegate.refFromURL(url);
+  }
+
+  goOffline(): void {
+    this._delegate.goOffline();
+  }
+
+  goOnline(): void {
+    this._delegate.goOnline();
+  }
+
+  _delete(): Promise<void> {
+    return this._delegate.INTERNAL.delete();
+  }
+
+  _setDatabaseUrl(url: string) {}
+}
+
+const ServerValue = Database.ServerValue;
+export { ServerValue };
+
+/**
+ * Initializes a new instance of the Realtime Database. If a custom database URL
+ * is specified, the URL overrides the database URL provided by the Firebase
+ * App.
+ *
+ * @param app - The {@link FirebaseApp} with which the `Firestore` instance will
+ * be associated.
+ * @param url - A settings object to configure the `Firestore` instance.
+ * @returns A newly initialized `Firestore` instance.
+ */
+export function initializeDatabase(
+  app: FirebaseApp,
+  url: string
+): FirebaseDatabase {
+  const firestore = _getProvider(
+    app,
+    'database-exp'
+  ).getImmediate() as FirebaseDatabase;
+  firestore._databaseUrl = url;
+  return firestore;
+}
+
+/**
+ * Returns the instance of the Realtime Database SDK that is associated
+ * with the provided {@link FirebaseApp}. Initializes a new instance with
+ * with default settings if no instance exists or if the existing instance uses
+ * a custom database URL.
+ *
+ * @param app - The {@link FirebaseApp} instance that the returned Firestore
+ * instance is associated with.
+ * @returns The `Firestore` instance of the provided app.
+ */
+export function getDatabase(app: FirebaseApp): FirebaseDatabase {
+  return _getProvider(app, 'database-exp').getImmediate() as FirebaseDatabase;
+}
```

Comment:
Should we just have `getDatabase(app: FirebaseApp, url?: string)`? This also removes the potential confusion you mentioned in the chat. Having a second optional parameter in `getFoo()` is the standard way to support multiple instances. see functions - https://github.com/firebase/firebase-js-sdk/blob/master/packages-exp/functions-exp/src/api.ts#L43

Reply:
That would work for me, but it seems inconsistent with Firebase Auth and Firestore. I can update this PR, but we should still talk about how we can consistently pass settings to our various SDKs. For example, I don't see us changing the `getFirestore()` call to accept setting since we wouldn't want users to specify the same settings over and over again. We also would likely not want to return a different instance if the settings are different.

Reply:
Agreed we need to standardize passing settings. My immediate proposal is use `getFoo(app, identifier?)` for getting instances with identifier and use `initializeFoo(app, identifier, settings)` for passing settings, which will throw if you try to initialize the same app + identifier twice.

Reply:
For this to work, should I add the `identifier` as an argument to `_getProvider()` or do you have a different solution in mind?

Reply:
No need. `provider.get()` and `provider.getImmeidate()` already support `identifier`.

Reply:
Awesome, thanks for the information. I updated the PR.